app/models/host/discovered.rb

```diff
@@ -133,22 +133,35 @@ def proxied?
     subnet.present? && subnet.discovery.present?
   end
 
+  def ip4or6
+    if Setting[:discovery_prefer_ipv6]
+      self.ip6 || self.ip
+    else
+      self.ip || self.ip6
+    end
+  end
+
   def proxy_url(node_ip)
-    proxied? ? subnet.discovery.url + "/discovery/#{node_ip}" : "https://#{node_ip}:8443"
+    wrapped_ip = node_ip =~ Resolv::IPv6::Regex ? "[#{node_ip}]" : node_ip
```

Comment:
Why don't you return a IPAddr object in `ip4or6` in the hope that this is not needed anymore.

Reply:
Just like that? (Amended)

Reply:
Yep!